pkg/catalog/prefix_iterator.go

```diff
@@ -13,7 +13,9 @@ type prefixIterator struct {
 }
 
 func NewPrefixIterator(it EntryIterator, prefix Path) EntryIterator {
-	it.SeekGE(prefix)
+	if len(prefix) > 0 {
+		it.SeekGE(prefix)
+	}
```

Comment:
Think this one can cause a bug - we assume we accept a fresh 'it' and we do not need to call SeekGE, but we must

Reply:
Agreed, fixed on the caller side.